hil-test/Cargo.toml

```diff
@@ -375,3 +373,6 @@ incremental      = false
 opt-level        = 3
 lto              = false # LTO (thin or fat) miscompiles some tests on RISC-V
 overflow-checks  = true
+
+[package.metadata.cargo-machete]
+ignored = ["semihosting"]
```

Comment:
Could we put a comment next to these as to why they're ignored? Otherwise we may never remove them from this list once they're there